src/main/java/ch/heigvd/dai/tcp/Server.java

```diff
@@ -93,6 +97,63 @@ public void list(BufferedReader in, BufferedWriter out, Path path) throws IOExce
     out.flush();
   }
 
+  /**
+   * Handles DELETE request and its given path.
+   * 
+   * @param path to delete
+   * @param out  The output where the result will be sent
+   * @throws IOException when unable to write to the socket output
+   */
+  public void delete(BufferedReader in, BufferedWriter out, Path path) throws IOException {
+
+    Path full_path = work_dir.resolve(path).normalize();
+    File file = full_path.toFile();
+
+    if (!file.exists()) {
+      out.write(String.valueOf(Errno.ENOENT) + EOT);
```

Comment:
When sending the exit status/response to the user, you should flush the socket.

Reply:
Temporarily added flush after every output but when the updated REPL and structure will be merged this will be replaced with the sendError method